embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/Embrace.kt

```diff
@@ -0,0 +1,477 @@
+@file:Suppress("DEPRECATION")
+
+package io.embrace.android.embracesdk
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.webkit.ConsoleMessage
+import io.embrace.android.embracesdk.internal.EmbraceInternalInterface
+import io.embrace.android.embracesdk.internal.Systrace
+import io.embrace.android.embracesdk.internal.api.SdkApi
+import io.embrace.android.embracesdk.network.EmbraceNetworkRequest
+import io.embrace.android.embracesdk.spans.EmbraceSpan
+import io.embrace.android.embracesdk.spans.EmbraceSpanEvent
+import io.embrace.android.embracesdk.spans.ErrorCode
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.sdk.logs.export.LogRecordExporter
+import io.opentelemetry.sdk.trace.export.SpanExporter
+
+/**
+ * Entry point for the SDK. This class is part of the Embrace Public API.
+ *
+ *
+ * Contains a singleton instance of itself, and is used for initializing the SDK.
+ */
+@SuppressLint("EmbracePublicApiPackageRule")
+public class Embrace private constructor(
+    internal var impl: EmbraceImpl = Systrace.traceSynchronous("embrace-impl-init") { EmbraceImpl() }
+) : SdkApi {
+
+    public companion object {
+
+        /**
+         * Singleton instance of the Embrace SDK.
+         */
+        private val instance: Embrace = Embrace()
+
+        /**
+         * Gets the singleton instance of the Embrace SDK.
+         *
+         * @return the instance of the Embrace SDK
+         */
+        @JvmStatic
+        public fun getInstance(): Embrace = instance
+    }
+
+    @Suppress("deprecation")
+    override fun start(context: Context) {
+        start(context, AppFramework.NATIVE)
+    }
+
+    @Deprecated("Use {@link #start(Context)} instead.")
+    override fun start(context: Context, appFramework: AppFramework) {
+        impl.start(
+            context,
+            appFramework
+        ) { _: io.embrace.android.embracesdk.internal.payload.AppFramework? -> null }
+    }
+
+    @Deprecated("Use {@link #start(Context)} instead. The isDevMode parameter has no effect.")
+    override fun start(context: Context, isDevMode: Boolean) {
+        start(context)
+    }
+
+    @Deprecated("Use {@link #start(Context)} instead. The isDevMode parameter has no effect.")
+    override fun start(context: Context, isDevMode: Boolean, appFramework: AppFramework) {
+        impl.start(
+            context,
+            appFramework
+        ) { _: io.embrace.android.embracesdk.internal.payload.AppFramework? -> null }
+    }
+
+    override val isStarted: Boolean
+        get() = impl.isStarted
+
+    override fun setAppId(appId: String): Boolean {
```

Comment:
Is this something we want to remove or at least deprecate? We have a lot of extra complexity because we allow this to happen, and I'm not sure it's that well tested - or used